tests/python_package_test/test_basic.py

```diff
@@ -499,6 +499,94 @@ def check_asserts(data):
     check_asserts(lgb_data)
 
 
+def test_dataset_construction_overwrites_user_provided_metadata_fields():
+
+    X = np.array([[1.0, 2.0], [3.0, 4.0]])
+
+    position=np.array([0.0, 1.0], dtype=np.float32)
```

Comment:
Minor style change
```suggestion
    position = np.array([0.0, 1.0], dtype=np.float32)
```

Reply:
totally support it, not having spaces was a mistake here. Thanks!